app/src/main/java/de/berlindroid/zepatch/MainActivity.kt

```diff
@@ -232,38 +232,33 @@ private fun PatchableDetail(
         Column(
             modifier = Modifier.padding(innerPadding),
         ) {
-            var currentMode by remember { mutableStateOf(COMPOSABLE) }
-
-            ProgressPills(imageBitmap, reducedImageBitmap, currentMode)
+            ProgressPills(uiState.imageBitmap, uiState.reducedImageBitmap, uiState.previewMode)
 
             WizardContent(
-                currentMode,
-                imageBitmap,
-                colorCount,
-                reducedImageBitmap,
-                reducedHistogram,
+                uiState.previewMode,
+                uiState.imageBitmap,
+                uiState.colorCount,
+                uiState.reducedImageBitmap,
+                uiState.reducedHistogram,
                 name,
-                onBitmapUpdated = { imageBitmap = it },
-                onColorCountUpdated = { colorCount = it },
-                onReducedUpdated = { img, histo ->
-                    reducedImageBitmap = img;reducedHistogram = histo
-                },
-                onEmbroideryUpdated = { data, preview ->
-                    embroideryData = data
-                    embroideryPreviewImage = preview
-                },
-                patchable
+                onBitmapUpdated = viewModel::updateBitmap,
+                onColorCountUpdated = viewModel::updateColorCount,
+                computeReducedBitmap = viewModel::computeReducedBitmap,
```

Comment:
Nit pick: should that also be `onComputeReducedBitmap`? Consistency with the other names, or does work differently, hence the difference in naming?

Reply:
for `onBitmapUpdated` and `onColorCountUpdated` we ask the composable to do something and it calls these methods when it completes the task.

on the other hand the `computeReducedBitmap` is an imperative action which we are asking and external function to perform, in this case the ViewModel.

BUT I see a nit myself, I am not consistently using color vs colour. very irritating, I will fix it.